components/automate-gateway/handler/data_lifecycle/data_lifecycle.go

```diff
@@ -0,0 +1,205 @@
+		eventFeedPurgeClient:  eventFeedPurgeClient,
+	}
+}
+
+// Server is the data lifecycle server handler implementation
+type Server struct {
+	ingestJobClient       ingest.JobSchedulerClient
+	ingestPurgeClient     data_lifecycle.PurgeClient
+	compliancePurgeClient data_lifecycle.PurgeClient
+	eventFeedPurgeClient  data_lifecycle.PurgeClient
+}
+
+// NOTE: services is currently unimplemented as the API's are not stable
+
+// GetStatus returns the aggregate status across all data lifecycle schedules
+// and configuration.
+func (s *Server) GetStatus(ctx context.Context, req *api.GetStatusRequest) (*api.GetStatusResponse, error) {
+	res := &api.GetStatusResponse{}
+	var err error
+
+	res.Infra, err = s.GetInfraStatus(ctx, &api.GetInfraStatusRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	res.Compliance, err = s.GetComplianceStatus(ctx, &api.GetComplianceStatusRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	res.EventFeed, err = s.GetEventFeedStatus(ctx, &api.GetEventFeedStatusRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	return res, nil
+}
+
+// SetConfig configures all data lifecycle jobs
+func (s *Server) SetConfig(ctx context.Context, req *api.SetConfigRequest) (*api.SetConfigResponse, error) {
+	res := &api.SetConfigResponse{}
+	var err error
+
+	if infra := req.GetInfra(); infra != nil {
+		_, err = s.SetInfraConfig(ctx, infra)
+		if err != nil {
+			return res, err
+		}
+	}
+
+	if compliance := req.GetCompliance(); compliance != nil {
+		_, err = s.SetComplianceConfig(ctx, compliance)
+		if err != nil {
+			return res, err
+		}
+	}
+
+	if eventFeed := req.GetEventFeed(); eventFeed != nil {
+		_, err = s.SetEventFeedConfig(ctx, eventFeed)
+		if err != nil {
+			return res, err
+		}
+	}
+
+	return res, nil
+}
+
+// Run runs all data lifecycle jobs
+func (s *Server) Run(ctx context.Context, req *api.RunRequest) (*api.RunResponse, error) {
+	res := &api.RunResponse{}
+	var err error
+
+	_, err = s.RunInfra(ctx, &api.RunInfraRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	_, err = s.RunCompliance(ctx, &api.RunComplianceRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	_, err = s.RunEventFeed(ctx, &api.RunEventFeedRequest{})
+	if err != nil {
+		return res, err
+	}
+
+	return res, nil
+}
+
+func purgeShowToJobStatus(purge *data_lifecycle.ShowResponse) *api.JobStatus {
+	return &api.JobStatus{
+		Name:           purge.InstanceName,
+		Disabled:       !purge.Enabled,
+		Recurrence:     purge.Recurrence,
+		NextDueAt:      purge.NextDueAt,
+		LastEnqueuedAt: purge.LastEnqueuedAt,
+		LastStartedAt:  purge.LastStart,
+		LastEndedAt:    purge.LastEnd,
+		LastElapsed:    maybeLastElapsed(purge.LastStart, purge.LastEnd),
+		PurgePolicies: &api.PurgePolicies{
+			Elasticsearch: purge.EsPolicies,
+			Postgres:      purge.PgPolicies,
+		},
+	}
+}
+
+func jobSettingsToPurgeConfigure(setting *api.JobSettings) *data_lifecycle.ConfigureRequest {
+	return &data_lifecycle.ConfigureRequest{
+		Enabled:    !setting.Disabled,
```

Comment:
was this deliberate? would it be safer to keep using `Enabled` (for example, forgetting to pass in disabled or typoing it could accidentally enable it

Reply:
Yeah, not using enabled is by design because of go's default values. It's much easier to handle disabled being defaulted to false than enable being defaulted to false. Rather than implicitly disabling things you have to explicitly disable them.